app/controllers/sessions_controller.rb

```diff
@@ -1,30 +1,35 @@
 class SessionsController < ApplicationController
-  
+  skip_before_action :authenticate_request, only: %i[create]
+
   def create
     @user = User.find_by(email: session_params[:email])
+    command = AuthenticateUser.call(session_params[:email], session_params[:password])
 
-    if @user && @user.authenticate(session_params[:password])
-      login!
-      render json: { logged_in: true, user: @user  }
+    if @user && command.success?
+      cookies[:token] = command.result
+      current_user = @user
+      render json: { logged_in: true, user: @user }
 
     else
-      render json: { 
-        status: :unauthorized, 
-        errors: ['no such user', 'verify credentials and try again or sign up'] 
+      render json: {
+        status: :unauthorized,
+        errors: ['no such user', 'verify credentials and try again or sign up']
       }
     end
   end
 
-  def is_logged_in?
-    if logged_in? && current_user
-      render json: { logged_in: true, user: current_user  }
+  def logged_in?
+    if current_user
+      render json: { logged_in: true, user: current_user }
     else
+      # TODO: : use refresh token here probably
       render json: { logged_in: false, message: 'no such user' }
```

Comment:
does this else case ever run? since the request presumably gets 401ed by the filter